src/include/quick-lint-js/lex.h

```diff
@@ -302,6 +302,7 @@ class lexer {
                                            error_reporter*);
 
   void parse_binary_number();
+  void parse_octal_number(bool strict, bool is_0o);
```

Comment:
Instead of bool parameters, how about an enum? Maybe something like this:

```cpp
enum class octal_kind {
  non_strict,
  //strict_0,
  strict_0o,
};
```

Reply:
This is probably better.  Depends I think on how many cases there really are for octals in strict and sloppy* mode.  For example, is there a difference in `0o` in strict and sloppy mode?  Not as for as I know, but if there were it could result in a lot of enumeration.  Which...actually, is the point of an enum anyways.

* While trying to learn about how a [particular piece](https://github.com/v8/v8/blob/master/src/parsing/parser.cc#L1545) of V8 works, I found out they call non_strict mode "sloppy", which I like